### PR TITLE
ci: better stateless runner (correctly collect artifacts and also some basic errors capturing)

### PR DESCRIPTION
**Pull request by azat**
**See commit messages**

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-ch-test-poll1** [comment]
*This is an automated comment for commit 15983c62364661ec7ec3e3e6d915a511789b3ada with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/67752/15983c62364661ec7ec3e3e6d915a511789b3ada/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/e4c165e919df7b43671fd8909176e2162c9a65fc/docs_check.html">✅ success</a></td></tr>
<tr><td>Fast test</td><td>Normally this is the first check that is ran for a PR. It builds ClickHouse and runs most of <a href="https://clickhouse.com/docs/en/development/tests#functional-tests">stateless functional tests</a>, omitting some. If it fails, further checks are not started until it is fixed. Look at the report to see which tests fail, then reproduce the failure locally as described <a href="https://clickhouse.com/docs/en/development/tests#functional-test-locally">here</a></td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/c629e2af50edbeb2b12f61f0a9b1a998251eec45/fast_test.html">✅ success</a></td></tr>
<tr><td>Unit tests</td><td>Runs the unit tests for different release types</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/c629e2af50edbeb2b12f61f0a9b1a998251eec45/unit_tests__release_.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

File: docker/test/stateless/run.sh [hunk 1]
```diff
@@ -416,4 +419,7 @@ if [[ "$USE_SHARED_CATALOG" -eq 1 ]]; then
     tar -chf /test_output/coordination1.tar /var/lib/clickhouse1/coordination ||:
 fi
 
+# Grep logs for sanitizer asserts, crashes and other critical errors
+check_logs_for_critical_errors
```
Comment (alexey-milovidov) [inline docker/test/stateless/run.sh:423]: I cannot find `source /stress_tests.lib`

Comment (azat) [reply docker/test/stateless/run.sh:430]: Nice catch, fixed.

**Review by alexey-milovidov — APPROVED** [review]

**Comment by azat** [comment]
@alexey-milovidov it had been merged too early, there was still an error, not a fatal, but because of this logs hadn't been processed completely, this should https://github.com/ClickHouse/ClickHouse/pull/67791 (and now it will detect if the script failed)

**Pull request merged**